core/controllers/topics_and_skills_dashboard.py

```diff
@@ -176,19 +178,23 @@ def post(self):
         skill.update_explanation(
             state_domain.SubtitledHtml.from_dict(explanation_dict))
 
-        filenames_in_skill = skill.get_all_image_filenames_during_creation()
-        if filenames != filenames_in_skill:
-            raise self.InvalidInputException(
-                'The filenames sent don\'t match the images in the skill.')
+        image_filenames = skill_services.get_image_filenames_from_skill(skill)
```

Comment:
Let's say that some error arises with the image validation and this controller returns an error. What happens to the image reference that's already in the skill? Does it just show a 404 image? Do we notify the creator?

Reply:
Currently, it would show a 404 image. Though, since the skill editor is opened right after this, the creator would get to know if some images had some issues there itself and can re-add them in the skill editor, right?

Reply:
Yeah, it would just show the 404 error, though the creator can change the image once they see that in the editor, right?

Reply:
I don't know, tbh. What would the actual UX look like? Maybe show screenshots/video?

If it's not intuitive then we need to draw attention to the issue. Also I don't think you can assume that the creator will immediately edit the item -- they may be creating skills in bulk and also they might think that after they've saved a skill that it's sufficiently complete.

Reply:
The skill editor is opened right after the skill is created, so wouldn't that be enough as any invalid images would be shown there.

Reply:
Done, showing an AlertsService warning if an error occurs along with the skill editor URL in it.